### ISSUE TITLE
Failed to fetch dependencies on Android: google-signin-support

### ISSUE THREAD

**Issue by Zorcher** [googlesamples/google-signin-unity#5]
I've created a Unity project with this plugin added, and regardless of running the sample or unpacking the unitypackage into my own project, I receive the following error when attempting to resolve dependencies on Android:

Resolution failed: Failed to fetch the following dependencies:
com.google.signin:google-signin-support:1+@aar

The project does build an APK, but running the app on the physical hardware and pressing the "Sign in with Google" button results in the following exception:

DllNotFoundException: native-googlesignin
  at (wrapper managed-to-native) Google.Impl.GoogleSignInImpl:GoogleSignIn_Create (intptr)
  at Google.Impl.GoogleSignInImpl..ctor (Google.GoogleSignInConfiguration configuration) [0x00000] in <filename unknown>:0
  at Google.GoogleSignIn.get_DefaultInstance () [0x00000] in <filename unknown>:0
  at LoginController.OnSignIn () [0x00000] in <filename unknown>:0

Is there a way for me to manually resolve the dependency?
On the bright side, I've built the IOS version and can confirm the login button works beautifully.

Thanks in advance.

**Comment by stewartmiles** [comment]
@Zorcher could you share the complete log of the dependency resolution failure on Android?

**Comment by Zorcher** [comment]
Sure, here's the complete log:

Resolution failed

Failed to fetch the following dependencies:
com.google.signin:google-signin-support:1+@aar


UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel)
GooglePlayServices.ResolverVer1_1:LogMissingDependenciesError(List`1)
GooglePlayServices.<DoResolutionUnsafe>c__AnonStorey18:<>m__26(List`1)
GooglePlayServices.<GradleResolution>c__AnonStorey14:<>m__1F(Result)
GooglePlayServices.<GradleResolution>c__AnonStorey15:<>m__28()
GooglePlayServices.PlayServicesResolver:PumpUpdateQueue()
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()

**Comment by stewartmiles** [comment]
@Zorcher you sure that's it?  We report a prior error message that describes why the dependency resolution failed.  From the dependency that [Android Resolver](https://github.com/googlesamples/unity-jar-resolver) failed to fetch, it looks like you could be missing https://github.com/googlesamples/google-signin-unity/tree/master/GoogleSignInPlugin/Assets/GoogleSignIn/Editor/m2repository/com/google/signin/google-signin-support from your project.

**Comment by Zorcher** [comment]
I've imported the google-signin-plugin-1.0.1.unitypackage into my project.  Here is the directory structure for the folder:

Directory of C:\UnityProjectFolder\Assets\GoogleSignIn\Editor\m2repository\com\google\signin\google-signin-support

11/06/2017  05:58 PM    <DIR>          .
11/06/2017  05:58 PM    <DIR>          ..
11/04/2017  02:06 AM    <DIR>          1.0.1
11/03/2017  10:20 AM               191 1.0.1.meta
11/03/2017  10:20 AM               318 maven-metadata.xml
11/03/2017  10:20 AM                32 maven-metadata.xml.md5
11/04/2017  02:04 AM               209 maven-metadata.xml.md5.meta
11/04/2017  02:04 AM               212 maven-metadata.xml.meta
11/03/2017  10:20 AM                40 maven-metadata.xml.sha1
11/04/2017  02:04 AM               209 maven-metadata.xml.sha1.meta
               8 File(s)          1,211 bytes

Directory of C:\UnityProjectFolder\Assets\GoogleSignIn\Editor\m2repository\com\google\signin\google-signin-support\1.0.1

11/06/2017  05:58 PM    <DIR>          .
11/06/2017  05:58 PM    <DIR>          ..
11/03/2017  10:20 AM                32 google-signin-support-1.0.1.aar.md5
11/04/2017  02:04 AM               188 google-signin-support-1.0.1.aar.md5.meta
11/03/2017  10:20 AM                40 google-signin-support-1.0.1.aar.sha1
11/04/2017  02:04 AM               188 google-signin-support-1.0.1.aar.sha1.meta
11/03/2017  10:20 AM               669 google-signin-support-1.0.1.pom
11/03/2017  10:20 AM                32 google-signin-support-1.0.1.pom.md5
11/04/2017  02:04 AM               188 google-signin-support-1.0.1.pom.md5.meta
11/04/2017  02:04 AM               188 google-signin-support-1.0.1.pom.meta
11/03/2017  10:20 AM                40 google-signin-support-1.0.1.pom.sha1
11/04/2017  02:04 AM               188 google-signin-support-1.0.1.pom.sha1.meta
              11 File(s)          1,753 bytes

**Comment by stewartmiles** [comment]
@Zorcher more of the log would be useful.  You could also turn on verbose logging under `Assets > Play Services Resolver > Android Resolver > Settings` which will help us narrow down the issue.

**Comment by Zorcher** [comment]
Of course, please find attached.

[unity-googlesignin-resolve-crash-dump.txt](https://github.com/googlesamples/google-signin-unity/files/1448416/unity-googlesignin-resolve-crash-dump.txt)

**Comment by stewartmiles** [comment]
Under `C:\UnityProjectFolder\Assets\GoogleSignIn\Editor\m2repository\com\google\signin\google-signin-support\1.0.1` you're missing `google-signin-support-1.0.1.aar` which is what the Android Resolver is complaining about in the log message:

```
Command completed: Successfully executed 'C:\UnityProjectFolder\Temp\PlayServicesResolverGradle\gradlew.bat -b "C:\UnityProjectFolder\Temp\PlayServicesResolverGradle\PlayServicesResolver.scripts.download_artifacts.gradle" --no-daemon "-PANDROID_HOME=C:/Users/Zorcher/AppData/Local/Android/sdk" "-PTARGET_DIR=C:\UnityProjectFolder\Assets\Plugins\Android" "-PMAVEN_REPOS=file:///C:/UnityProjectFolder/Assets/GoogleSignIn/Editor/m2repository" "-PPACKAGES_TO_COPY=com.google.signin:google-signin-support:1+;com.google.android.gms:play-services-auth:10+"'
stdout:
:copyPackages
Copied artifacts:
play-services-auth-10.2.6.aar
play-services-auth-base-10.2.6.aar
play-services-base-10.2.6.aar
play-services-tasks-10.2.6.aar
play-services-basement-10.2.6.aar
support-v4-25.2.0.aar
support-fragment-25.2.0.aar
support-media-compat-25.2.0.aar
support-core-utils-25.2.0.aar
support-core-ui-25.2.0.aar
support-compat-25.2.0.aar
support-annotations-25.2.0.jar

Missing artifacts:
com.google.signin:google-signin-support:1+@aar


BUILD SUCCESSFUL in 5s
1 actionable task: 1 executed

stderr:

exit code: 0
```

@claywilkinson I just noticed that we're including the `.aar` in the m2repository folder rather than renaming the artifact `.srcaar`.  Putting a `.aar` file anywhere in newer Unity projects results in it being included in the build unfortunately.  So the Android Resolver hacks around this by allowing the user to store a `.srcaar` artifact which is renamed to `.aar` when it's resolved.  Any chance you could fix this?

**Comment by claywilkinson** [comment]
Yep I will today

**Comment by Zorcher** [comment]
Is there an ETA on a fix date or a manual way to solve the dependency?

**Comment by claywilkinson** [comment]
Working on it :)   A manual workaround is to 
1. rename **Assets/GoogleSignIn/Editor/m2repository/com/google/signin/google-signin-support/1.0.1/google-signin-support-1.0.1.aar** to have the extension __.srcaar__
2. update the metadata in **Assets/GoogleSignIn/Editor/m2repository/com/google/signin/google-signin-support/1.0.1/google-signin-support-1.0.1.pom** to have ` <packaging>srcaar</packaging>`

**Comment by Zorcher** [comment]
Is there a manual download link for the google-signin-support-1.0.1.aar file?

**Comment by mrttson** [comment]
I got the same issues. Maybe unitypackage missing "google-signin-support-1.0.1.aar"?
I got these files after import unitypackage.

**..\Assets\GoogleSignIn\Editor\m2repository\com\google\signin\google-signin-support\1.0.1**
![capture](https://user-images.githubusercontent.com/4798948/32697170-588bf288-c7bd-11e7-9f17-2bd55a2ac874.PNG)

**Comment by claywilkinson** [comment]
This is fixed in version 1.0.2 of the plugin.
